binding/vue/README.md

```diff
@@ -1,127 +1,250 @@
+```
+
+### Base64
+
+**NOTE**: This method works without hosting a server, but increases the size of the model file roughly by 33%.
+
+This method uses a base64 string of the model file and feeds it to Porcupine. Use the built-in script `pvbase64` to
+base64 your model file:
+
+```console
+npx pvbase64 -i ${PORCUPINE_MODEL_FILE} -o ${OUTPUT_DIRECTORY}/${MODEL_NAME}.js
+```
+
+The output will be a js file which you can import into any file of your project. For detailed information
+about `pvbase64`,
+run:
+
+```console
+npx pvbase64 -h
+```
+
+### Porcupine Model
+
+Porcupine saves and caches your parameter model file (`.pv`) in IndexedDB to be used by Web Assembly.
+Use a different `customWritePath` variable to hold multiple model values and set the `forceWrite` value to true to force
+re-save the model file.
+If the model file changes, `version` should be incremented to force the cached models to be updated.
+Either `base64` or `publicPath` must be set to instantiate Porcupine. If both are set, Porcupine will use the `base64`
+model.
+
+```typescript
+// Model (.pv)
+const porcupineModel = {
+  publicPath: ${MODEL_RELATIVE_PATH},
+  // or
+  base64: ${MODEL_BASE64_STRING},
+
+  // Optional
+  customWritePath: 'custom_model',
+  forceWrite: true,
+  version: 1,
+}
+```
+
+### Initialize Porcupine
+
+**Note**: Due to limitations on Vue, one component can only have one instance of Porcupine. To use multiple 
+instance of Porcupine instead, check out [Porcupine Web binding](https://www.npmjs.com/package/@picovoice/porcupine-web).
+
+Create the following functions:
+ - `keywordDetectionCallback` function to get the streaming results from the worker:
+ - `isLoadedCallback` function to check if Porcupine has loaded:
+ - `isListeningCallback` function to check if Porcupine started/stopped listening for wake word detection:
+
+```typescript
+...
+methods: {
+  keywordDetectionCallback: function(keyword) {
+    console.log(`Porcupine detected keyword: ${keyword.label}`);
   },
-  methods: {
-    start: function () {
-      if (this.$porcupine.start()) {
-        this.isListening = !this.isListening;
-      }
-    },
-    pause: function () {
-      if (this.$porcupine.pause()) {
-        this.isListening = !this.isListening;
-      }
-    },
-    ppnReadyFn: function() {
-      this.isLoaded = true;
-    },
-    ppnKeywordFn: function(data: string) {
-      this.detections = [...this.detections, data.keywordLabel];
-    },
-    ppnErrorFn: function(error: Error) {
-      this.isError = true;
-      this.errorMessage = error.toString();
-    },
+  isLoadedCallback: function(isLoaded) {
+    console.log(isLoaded);
   },
+  isListeningCallback: function(isListening) {
+    console.log(isListening);
+  }
 };
+...
+```
+
+Add to the `options` object an `processErrorCallback` function if you would like
+to catch errors:
+
+```typescript
+methods: {
+  processErrorCallback: function (error: string) {
+  ...
+  }
+}
+
```

Comment:
Is this necessary to mention since we already include a way to get the errors through the sdk?